metafields.go

```diff
@@ -0,0 +1,196 @@
+	// ListVolumeMetaFieldType is a list of values and a unit of volume. Valid unit values: ml, cl, l, m3 (cubic meters), us_fl_oz, us_pt, us_qt, us_gal, imp_fl_oz, imp_pt, imp_qt, imp_gal.
+	ListVolumeMetaFieldType MetafieldListType = "list.volume"
+	// ListWeightMetaFieldType is a list of values and a unit of weight. Valid unit values: oz, lb, g, kg
+	ListWeightMetaFieldType MetafieldListType = "list.weight"
+)
+
+// Resources
+const (
+	// ArticleResource is the resource for metafields related to articles.
+	ArticleResource OwnerResource = "article"
+	// BlogResource is the resource for metafields related to the blog.
+	BlogResource OwnerResource = "blog"
+	// CollectionResource is the resource for metafields related to collections.
+	CollectionResource OwnerResource = "collection"
+	// CustomerResource is the resource for metafields related to customers.
+	CustomerResource OwnerResource = "customer"
+	// DraftOrderResource is the resource for metafields related to draft orders.
+	DraftOrderResource OwnerResource = "draft_order"
+	// LocationResource is the resource for metafields related to locations.
+	LocationResource OwnerResource = "location"
+	// OrderResource is the resource for metafields related to orders.
+	OrderResource OwnerResource = "order"
+	// PageResource is the resource for metafields related to pages.
+	PageResource OwnerResource = "page"
+	// ProductResource is the resource for metafields related to products.
+	ProductResource OwnerResource = "product"
+	// ProductImageResource is the resource for metafields related to product images.
+	ProductImageResource OwnerResource = "product_image"
+	// ProductVariantResource is the resource for metafields related to product variants.
+	ProductVariantResource OwnerResource = "variants"
+	// ShopResource is the resource for metafields related to the shop.
+	ShopResource OwnerResource = "shop"
+)
+
+// Metafield is a flexible way to attach additional information to a Shopify resource (e.g. Product, Collection, etc.).
+type Metafield struct {
+	// ID is the unique ID of the metafield.
+	ID int64
+	// Description is the description of the information that the metafield contains.
+	Description string
+	// Key is the key of the metafield. Keys can be up to 30 characters long and can contain alphanumeric characters, hyphens, underscores, and periods.
+	Key string
+	// Namespace is the container name for a group of metafields. Grouping metafields within a namespace prevents your metafields from conflicting with other metafields with the same key name. Must have between 3-20 characters.
+	Namespace string
+	// Resource is the unique ID of the resource that the metafield is attached to and the type of resource that the metafield is attached to.
+	Resource MetafieldResource
+	// Value is the data stored in the metafield. The value is always stored as a string, regardless of the metafield's type.
+	Value string
+	// MetafieldType is the type of data that the metafield stores in the `value` field.
```

Comment:
I assume all of these comments were just taken from the Shopify documentation? Should this not be `Type`?
```suggestion
	// Type is the type of data that the metafield stores in the `value` field.
```

Reply:
I called it MetafieldType because all these repositories exist within the Shopify package so it's being specific as a MetafieldType.